### PR TITLE
April Finance Update

### PR DESCRIPTION
**Pull request by efecarranza**
# Changelog

Note: Includes BGDs reimbursements.

### CODE REVIEW

File: src/20240421_Multi_AprilFinanceUpdate/AprilFinanceUpdate.md
Comment (brotherlymite) [inline src/20240421_Multi_AprilFinanceUpdate/AprilFinanceUpdate.md:84]:
```suggestion
```

File: src/20240421_Multi_AprilFinanceUpdate/AaveV2Ethereum_AprilFinanceUpdate_20240421.t.sol
Comment (ianflexa) [inline src/20240421_Multi_AprilFinanceUpdate/AaveV2Ethereum_AprilFinanceUpdate_20240421.t.sol:40]:
```suggestion

function test_defaultProposalExecution() public {
    defaultTest(
      'AaveV2Ethereum_AprilFinanceUpdate_20240421',
      AaveV2Ethereum.POOL,
      address(proposal)
    );
  }
  
```

File: src/20240421_Multi_AprilFinanceUpdate/AaveV2Polygon_AprilFinanceUpdate_20240421.t.sol
Comment (ianflexa) [inline src/20240421_Multi_AprilFinanceUpdate/AaveV2Polygon_AprilFinanceUpdate_20240421.t.sol:37]:
```suggestion

function test_defaultProposalExecution() public {
    defaultTest(
      'AaveV2Polygon_AprilFinanceUpdate_20240421',
      AaveV2Polygon.POOL,
      address(proposal)
    );
  }

```

File: src/20240421_Multi_AprilFinanceUpdate/AaveV3Gnosis_AprilFinanceUpdate_20240421.t.sol
Comment (ianflexa) [inline src/20240421_Multi_AprilFinanceUpdate/AaveV3Gnosis_AprilFinanceUpdate_20240421.t.sol:21]:
```suggestion

function test_defaultProposalExecution() public {
    defaultTest(
      'AaveV3Gnosis_AprilFinanceUpdate_20240421',
      AaveV3Gnosis.POOL,
      address(proposal)
    );
  }
  
```

**Review by ianflexa — CHANGES_REQUESTED** [review]
missing e2e tests

File: src/20240421_Multi_AprilFinanceUpdate/AaveV2Ethereum_AprilFinanceUpdate_20240421.sol
Comment (brotherlymite) [inline src/20240421_Multi_AprilFinanceUpdate/AaveV2Ethereum_AprilFinanceUpdate_20240421.sol:81]: @efecarranza I would add an if condition here, to approve the GHO amount only when the allowance of aUSDT to AGD is 612.9k as if claimed mistakenly before the payload execution, the transfer could be done twice.

File: src/20240421_Multi_AprilFinanceUpdate/AprilFinanceUpdate.md
Comment (brotherlymite) [inline src/20240421_Multi_AprilFinanceUpdate/AprilFinanceUpdate.md:26]:
```suggestion
Also within this proposal, there is the corrective action of returning [3504 RealT WXDAI](https://gnosisscan.io/tx/0xf70a507847da2ad4acbecc35cd84c2f5d2489b0b0eb0e16af49c9262ca95707e) sent by RealT to the Aave Gnosis Chain collector by accident. Details on the agreement with realT can be found [here](https://snapshot.org/#/aave.eth/proposal/0xff69be7580614ebc1a455591c1bd651d8f0af12070d277d7d8846beb3c7c964b).
```

File: src/20240421_Multi_AprilFinanceUpdate/AprilFinanceUpdate.md
Comment (brotherlymite) [inline src/20240421_Multi_AprilFinanceUpdate/AprilFinanceUpdate.md:24]:
```suggestion
This proposal cancels the existing USDT Allowance to the Aave Grants DAO and replaces it with a GHO Allowance of equal size, 612.9k. This represents a continuation of transitioning to using GHO as the preferred means of covering DAO expenses.
```

File: src/20240421_Multi_AprilFinanceUpdate/AprilFinanceUpdate.md
Comment (brotherlymite) [inline src/20240421_Multi_AprilFinanceUpdate/AprilFinanceUpdate.md:71]: I would put an approximate amount instead of (All - 1)

File: src/20240421_Multi_AprilFinanceUpdate/AaveV2Ethereum_AprilFinanceUpdate_20240421.sol
Comment (efecarranza) [reply src/20240421_Multi_AprilFinanceUpdate/AaveV2Ethereum_AprilFinanceUpdate_20240421.sol:81]: @brotherlymite sorry say that again, i don't think i understood.

File: src/20240421_Multi_AprilFinanceUpdate/AprilFinanceUpdate.md
Comment (efecarranza) [reply src/20240421_Multi_AprilFinanceUpdate/AprilFinanceUpdate.md:71]: we've been using this terminology in all these proposals, if it's okay i'd rather keep this way.

**Comment by efecarranza** [comment]
@brotherlymite @ianflexa 

ian, i added the tests but when they yield empty files we don't include them usually.
harsh, updated but have one question.

thanks

File: src/20240421_Multi_AprilFinanceUpdate/AprilFinanceUpdate.md
Comment (brotherlymite) [reply src/20240421_Multi_AprilFinanceUpdate/AprilFinanceUpdate.md:71]: okay

File: src/20240421_Multi_AprilFinanceUpdate/AaveV2Ethereum_AprilFinanceUpdate_20240421.sol
Comment (brotherlymite) [reply src/20240421_Multi_AprilFinanceUpdate/AaveV2Ethereum_AprilFinanceUpdate_20240421.sol:81]:
```
if (USDT_A_TOKEN.allowance(COLLECTOR, AGD_MULTISIG) >= ~612.9e6) {
  AaveV3Ethereum.COLLECTOR.approve(
      AaveV3EthereumAssets.GHO_UNDERLYING,
      AGD_MULTISIG,
      AGD_GHO_ALLOWANCE
    );
}
```
@efecarranza if the AGD wallet claims aUSDT before proposal execution, we still approve the GHO token. So there is a double approval. So just an extra check.

File: src/20240421_Multi_AprilFinanceUpdate/AaveV2Polygon_AprilFinanceUpdate_20240421.sol
Comment (brotherlymite) [inline src/20240421_Multi_AprilFinanceUpdate/AaveV2Polygon_AprilFinanceUpdate_20240421.sol:131]:
```suggestion
```
@efecarranza I would not send funds to the plasma bridge, as we will do the MATIC bridging in second part.

File: src/20240421_Multi_AprilFinanceUpdate/AaveV2Polygon_AprilFinanceUpdate_20240421.sol
Comment (brotherlymite) [inline src/20240421_Multi_AprilFinanceUpdate/AaveV2Polygon_AprilFinanceUpdate_20240421.sol:44]:
```suggestion
```

File: src/20240421_Multi_AprilFinanceUpdate/AaveV2Polygon_AprilFinanceUpdate_20240421.sol
Comment (brotherlymite) [inline src/20240421_Multi_AprilFinanceUpdate/AaveV2Polygon_AprilFinanceUpdate_20240421.sol:21]:
```suggestion
```

File: src/20240421_Multi_AprilFinanceUpdate/AaveV2Polygon_AprilFinanceUpdate_20240421.sol
Comment (brotherlymite) [inline src/20240421_Multi_AprilFinanceUpdate/AaveV2Polygon_AprilFinanceUpdate_20240421.sol:97]: not needed now, as the we would do the matic bridging in the second part

File: src/20240421_Multi_AprilFinanceUpdate/AaveV2Polygon_AprilFinanceUpdate_20240421.sol
Comment (Luigy-Lemon) [reply src/20240421_Multi_AprilFinanceUpdate/AaveV2Polygon_AprilFinanceUpdate_20240421.sol:131]: Just one remark, Polygon plasma bridge has a log cap protection of 10 logs in a bridge transaction request.  Transferring funds to the bridge contract in Part A makes sense to me to avoid this issue during Part B.  

Currently not sure if it will be a problem or not since this is the first time we are doing it in production.

File: src/20240421_Multi_AprilFinanceUpdate/AaveV2Polygon_AprilFinanceUpdate_20240421.sol
Comment (brotherlymite) [inline src/20240421_Multi_AprilFinanceUpdate/AaveV2Polygon_AprilFinanceUpdate_20240421.sol:180]:
```suggestion
```
@efecarranza This is not needed as `address(this)` is the executor in this case which will receive the MATIC token.

File: src/20240421_Multi_AprilFinanceUpdate/AprilFinanceUpdate_20240421.s.sol
Comment (brotherlymite) [inline src/20240421_Multi_AprilFinanceUpdate/AprilFinanceUpdate_20240421.s.sol:56]:
```suggestion
    actionsTwo[0] = GovV3Helpers.buildAction(payload1);
```

File: src/20240421_Multi_AprilFinanceUpdate/AprilFinanceUpdate_20240421.s.sol
Comment (brotherlymite) [inline src/20240421_Multi_AprilFinanceUpdate/AprilFinanceUpdate_20240421.s.sol:112]:
```suggestion
    actionsPolygonTwo[0] = GovV3Helpers.buildAction(
```

File: src/20240421_Multi_AprilFinanceUpdate/AprilFinanceUpdate.md
Comment (ianflexa) [inline src/20240421_Multi_AprilFinanceUpdate/AprilFinanceUpdate.md:78]:
```suggestion
|   aEthDAI (All-1)      |                      |  aPolUSDC(All-1)  |                      |
```

File: src/20240421_Multi_AprilFinanceUpdate/AprilFinanceUpdate.md
Comment (brotherlymite) [inline src/20240421_Multi_AprilFinanceUpdate/AprilFinanceUpdate.md:82]:
```suggestion
Assets migrated from Polygon to Ethereum are to be deposited into Aave v3 where practical, with the exception of DAI and USDC which will be swapped to GHO during Part B.
```

**Review by ianflexa — COMMENTED** [review]
missing aEthDAI

**Review by ianflexa — DISMISSED** [review]

**Review by brotherlymite — APPROVED** [review]

**Review by ianflexa — APPROVED** [review]

**Pull request merged**
